### PR TITLE
Fix little problem with type hints and adopt a test to make sure it works in all platform

### LINKED ISSUE #1205: Little problem with the type hint
image
sqlalchemy does not exporse orm as a clear attr, so the type hint there is not working.
now we change that to this with full path.
image
And it works out with the type hint.

The main problem with that might be the use of some thing like `db.session`, which I use a lot in project to do CRUD operations. Previously we have that. It will stock since sa don't have orm attr. It's a module.

https://github.com/pallets-eco/flask-sqlalchemy/blob/d137db10e733daf1fd2ad24dd59aedc3072d48e5/src/flask_sqlalchemy/extension.py#L325

And I think we should make this little change to provide more type hint info.

Environment:

- Python version: 3.9.16
- Flask-SQLAlchemy version: 3.0.3
- SQLAlchemy version: 2.0.0 and above

### PR DESCRIPTION
**Pull request by Zoupers**
- fixes #1205
- fixes a test that cannot work on windows

Checklist:

- [x] ~~Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.~~
- [x] ~~Add or update relevant docs, in the docs folder and in code.~~
- [x] ~~Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.~~
- [x] ~~Add `.. versionchanged::` entries in any relevant code docs.~~
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

Delete line means I think there is no need to do that.

### CODE REVIEW

**Comment by pamelafox** [comment]
@Zoupers Is this ready for review? And can you specify in the description or inline, which test is the one that didn't work on Windows?

File: src/flask_sqlalchemy/query.py [hunk 1]
```diff
@@ -2,16 +2,15 @@
...
+import sqlalchemy.exc as sa_exc
+import sqlalchemy.orm as sa_orm
 from flask import abort
 
 from .pagination import Pagination
 from .pagination import QueryPagination
 
 
-class Query(sa.orm.Query):  # type: ignore[type-arg]
+class Query(sa_orm.Query[t.Any]):
```
Comment (pamelafox) [inline src/flask_sqlalchemy/query.py:13]: This seems correct per https://github.com/sqlalchemy/sqlalchemy/blob/1c3dbf39ec2d72ea70c0f74fa2e6dc537e76700b/lib/sqlalchemy/orm/query.py#L151

File: tests/test_record_queries.py [hunk 2]
```diff
@@ -24,5 +26,5 @@ class Example(db.Model):
     assert "FROM example" in info.statement
     assert info.parameters[0][0] == 5
     assert info.duration == info.end_time - info.start_time
-    assert "tests/test_record_queries.py:" in info.location
+    assert os.path.join("tests", "test_record_queries.py:") in info.location
```
Comment (pamelafox) [inline tests/test_record_queries.py:29]: I assume this is the test that did not previously work in Windows.

File: src/flask_sqlalchemy/query.py (hunk 1 above)
Comment (Zoupers) [reply src/flask_sqlalchemy/query.py:13]: According to this, https://github.com/sqlalchemy/sqlalchemy/blob/4808acfe1b88854fce47b792ef9995f1921426e7/lib/sqlalchemy/orm/query.py#L77
which is the version(`1.4.18`) specified in the `requirements/tests-min.in`, it's not correct. I will fix it.

**Comment by Zoupers** [comment]
I have fixed the bug and passed the corresponding tests locally. I apologize for any inconvenience caused.

**Comment by pamelafox** [comment]
@Zoupers Thanks! I think we may upgrade min version of SQLAlchemy to 2.0 in the next version in order to formally support it, but for this minor release, it's better to just get it working for 1.4. This is looking good now.

**Comment by pamelafox** [comment]
I experimented a bit to see if we could automatically test that this works correctly, but we're not currently running pyright on the code, and if we did, we'd have a bunch of other issues that can't be ignored as easily as they are in mypy (as pyright doesn't have the same specific error codes), so I think it's fine to not have a test for the pyright-specific typing right now.

However, if there's a regression in the future, we might consider adding pyright env to tox.

**Pull request merged**